lua/nvim-paredit/lang/clojure.lua

```diff
@@ -36,6 +36,16 @@ function M.get_node_root(node)
   return traversal.find_root_element_relative_to(root, node)
 end
 
+function M.element_lit(node)
+  local lit = node
+  if lit then
+    if not lit:type():match("lit$") then
+      lit = M.element_lit(lit:parent())
+    end
+  end
+  return lit
+end
+
```

Comment:
From what I can see this is trying to solve the same problem as the existing `get_not_root` fn. Unless I am mistaken, we don't need to introduce this new language API and can just use the existing fn.

To clarify this fn will find the root most node relative to another node.

`get_node_root(keyword_node, relative_parent)` -> `namespaced_keyword`.

Reply:
`:foo/my-keyword` is represended as
- kwd_lit `:` - whole node range
  - kwd_ns `foo` - namespace token range
  - kwd_name `my-keyword` - name token tange

And depending on where is your cursor it returns on of this three nodes.
say if our cursor is on ns or name, then `get_node_root` returns `kwd_lit` and we are fine. But it won't work if our cursor is on `:`

Reply:
`:foo/my-keywo|rd` - node is `kwd_name`
`:fo|o/my-keyword` - node is `kwd_ns`
`|:foo/my-keywo|rd` - node is `kwd_lit`

`element_lit` always returns `kwd_lit`

Reply:
Here `(:foo/my-ke|yword)` `element_lit` returns `kwd_lit`, but not the parent node which is parenthesis

Reply:
If the cursor is on `:` (`kwd_lit`) then `get_node_root` _should_ return `kwd_lit`. If not then it's a bug. Or perhaps I misunderstand you/the requirement

Reply:
`get_node_root` should also always return `kwd_lit` in all your above examples.

Reply:
Ok, you are right, I didn't get it from the description of `get_node_root` and was too lazy to really test stuff with it :). Removing `element_lit`.

Reply:
Yea the fn comment isn't particularly clear on what it's for/does haha. Cool